backend/classification/file_loading.py

```diff
@@ -41,16 +41,15 @@ def get_raw_array(file):
     Returns:
     - mne.RawArray of the two EEG channels of interest
     """
-    lines = file.readlines()
-    eeg_raw = np.zeros((len(lines) - SKIP_ROWS, len(EEG_CHANNELS)))
 
-    for index, line in enumerate(lines[SKIP_ROWS:]):
-        line_splitted = line.decode('utf-8').split(',')
+    retained_columns = tuple(range(1, len(EEG_CHANNELS) + 1))
+    eeg_raw = pd.read_csv(file,
+                          skiprows=SKIP_ROWS,
+                          usecols=retained_columns
+                          ).to_numpy()
 
-        if len(line_splitted) < CYTON_TOTAL_NB_CHANNELS:
-            raise ClassificationError()
-
-        eeg_raw[index] = _get_decimals_from_hexadecimal_strings(line_splitted)
+    hexstr_to_int = np.vectorize(_hexstr_to_int)
```

Comment:
Le problème avec `np.vectorize()` c'est qu'au final c'est un simple for loop (voir (docu)[https://numpy.org/doc/stable/reference/generated/numpy.vectorize.html] `The vectorize function is provided primarily for convenience, not for performance. The implementation is essentially a for loop.`)

Peut-être que [pd.Dataframes.apply()](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.apply.html) serait plus rapide?

Reply:
Dans ce cas-ci, apply ne s'applique pas, alors la meilleure alternative c'est applymap, mais les performances sont moins bonnes.

Reply:
La limitation ici est la fonction `_hexstr_to_int` :(